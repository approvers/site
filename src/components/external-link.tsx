import type { ReactNode } from "react";
import styles from "../scss/components/external-link.module.scss";

export interface ExternalLinkProps {
  href: string;
  children: ReactNode;
}

export const ExternalLink = (props: ExternalLinkProps): JSX.Element => (
  <a href={props.href} className={styles.link} rel="noopener noreferrer" target="_blank">
    {props.children}
  </a>
);

export const ExternalIconLink = (props: ExternalLinkProps): JSX.Element => (
  <a href={props.href} className={styles.iconLink} rel="noopener noreferrer" target="_blank">
    {props.children}
  </a>
);
