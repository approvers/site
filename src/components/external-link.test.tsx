import { ExternalIconLink, ExternalLink } from "./external-link";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(<ExternalLink href="https://duckduckgo.com">Go to DuckDuckGo</ExternalLink>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders icon correctly", () => {
  const tree = renderer
    .create(<ExternalIconLink href="https://duckduckgo.com">Go to DuckDuckGo</ExternalIconLink>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
