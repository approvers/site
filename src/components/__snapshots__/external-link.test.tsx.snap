// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="link"
  href="https://duckduckgo.com"
  rel="noopener noreferrer"
  target="_blank"
>
  Go to DuckDuckGo
</a>
`;

exports[`renders icon correctly 1`] = `
<a
  className="iconLink"
  href="https://duckduckgo.com"
  rel="noopener noreferrer"
  target="_blank"
>
  Go to DuckDuckGo
</a>
`;
